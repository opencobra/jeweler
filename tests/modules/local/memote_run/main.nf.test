nextflow_process {

    name 'Test Process MEMOTE_RUN'
    script 'modules/local/memote_run/main.nf'
    process 'MEMOTE_RUN'
    tag 'modules'
    tag 'modules_local'
    tag 'memote_run'

    test('Should succeed for a valid SBML document') {
        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of(
                    [[id: 'e_coli_core'], file('https://raw.githubusercontent.com/opencobra/jeweler/testdata/data/model/e_coli_core.xml.gz')]
                )
                """
            }
        }

        then {
            assert process.success

            with(process.out.report) {
                assert size() == 1

                with(get(0)) {
                    assert get(1).endsWith('.json.gz')
                    assert file(get(1)).size() > 0
                }
            }

            // Reported versions should match snapshots.
            assert snapshot(process.out.versions.first()).match('versions')
        }
    }

    test('Should fail for an invalid SBML document') {
        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = Channel.of(
                    [[id: 'e_coli_invalid'], file('https://raw.githubusercontent.com/opencobra/jeweler/testdata/data/model/e_coli_invalid.xml.gz')]
                )
                """
            }
        }

        then {
            assert process.failed
        }
    }
}
