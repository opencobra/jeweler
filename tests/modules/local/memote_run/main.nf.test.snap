{
    "versions": {
        "content": [
            "versions.yml:md5,d265b13e60ee9cdb2f71f1dffcbea5d0"
        ],
        "timestamp": "2023-11-28T16:44:37.742661143"
    },
    "Should succeed for a valid SBML document": {
        "content": [
            "e_coli_core.json.gz:md5,b2e484bf002b95ea39a3711262ccc0ac"
        ],
        "timestamp": "2023-11-28T16:45:11.794158498"
    }
}